### PR TITLE
feat(sdk): Remove timeout since we moved to default unlimited timeout

### PR DESCRIPTION
**Pull request by Tomcli**
**Which issue is resolved by this Pull Request:** 
Resolves #

**Description of your changes:**
Remove timeout since we moved to default unlimited timeout in #1042. Plus not all custom task included timeout support, so it make sense to reduce the yaml size and force our deployer to update the global timeout.

**Environment tested:**

* Python Version (use `python --version`):
* Tekton Version (use `tkn version`):
* Kubernetes Version (use `kubectl version`):
* OS (e.g. from `/etc/os-release`):

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: sdk/python/kfp_tekton/compiler/compiler.py [hunk 1]
```diff
@@ -1394,13 +1388,10 @@ def get_when_task(input_task_when, depended_conditions):
 
     # add workflow level timeout to pipeline run
     if not TEKTON_GLOBAL_DEFAULT_TIMEOUT or pipeline.conf.timeout:
```
Comment (yhwang) [inline sdk/python/kfp_tekton/compiler/compiler.py:1390]: looks like we don't need `TEKTON_GLOBAL_DEFAULT_TIMEOUT` anymore, should we also remove it?

Comment (Tomcli) [reply sdk/python/kfp_tekton/compiler/compiler.py:1390]: sure let me remove that

Comment (Tomcli) [reply sdk/python/kfp_tekton/compiler/compiler.py]: done

**Review by yhwang — APPROVED** [review]
/lgtm

**Pull request merged**